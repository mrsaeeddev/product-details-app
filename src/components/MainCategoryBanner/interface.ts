export interface MainCategoryProps {
    name: string;
}