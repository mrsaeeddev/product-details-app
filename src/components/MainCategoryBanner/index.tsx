import React from "react";
import { View, Text, StyleSheet, ImageBackground } from "react-native";
import { FONT_SIZE_H1, FONT_WEIGHT } from "../../constants/styling";
import { globalStyles } from "../../globalStyles";
import { MainCategoryProps } from "./interface";

const styles = StyleSheet.create({
    coverImage: {
        height: 300,
        resizeMode: 'stretch',
        borderRadius: 20,
    },
    heading: {
        fontSize: FONT_SIZE_H1,
        fontWeight: FONT_WEIGHT,
    }
});

function MainCategoryBanner(props: MainCategoryProps) {
    return (<View style={globalStyles.container}>
        <ImageBackground
            style={[styles.coverImage, globalStyles.alignItemsCenter]}
            imageStyle={{ borderRadius: 20 }}
            source={require('../../images/personalcare.jpg')}
        >
            <Text style={[globalStyles.textCenter, styles.heading]}>{props.name}</Text>
        </ImageBackground>
    </View>)
}

export default MainCategoryBanner;