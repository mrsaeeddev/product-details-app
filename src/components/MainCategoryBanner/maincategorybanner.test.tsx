import React from 'react';
import renderer from 'react-test-renderer';

import MainCategoryBanner from './';

describe('MainCategoryBanner', () => {
    it('should render correctly using Snapshot', () => {
        const name = "dummy"
        const tree = renderer.create(<MainCategoryBanner name={name} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});