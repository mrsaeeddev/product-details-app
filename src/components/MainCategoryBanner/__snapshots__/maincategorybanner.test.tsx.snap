// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainCategoryBanner should render correctly using Snapshot 1`] = `
<View
  style={
    Object {
      "padding": 20,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "borderRadius": 20,
          "height": 300,
          "resizeMode": "stretch",
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          Object {
            "borderRadius": 20,
          },
        ]
      }
    />
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
          },
        ]
      }
    >
      dummy
    </Text>
  </View>
</View>
`;
